### PR TITLE
Add support for Azure Blob Storage connection string authentication

### PR DESCRIPTION
**Pull request by suzusuzu**
### Motivation and context

This is a change to support Connection String authentication for Azure Blob Storage. For example, Connection String Authentication allows for Azure Blob Storage running on the IoT Edge of an edge device.

### How has this been tested?

I tested it by actually running it as shown in the next image, I could not find where the E2E test for cloudstorage was done.
cvat_blob_connection_string

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added a description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [x] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2022 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by Marishka17** [comment]
Hi, @suzusuzu, thanks for the contribution! Are you still going to contribute to us? If so, could you please resolve conflicts in this PR?

**Comment by suzusuzu** [comment]
@Marishka17 Thanks for the reply, I resolved the conflict. Please check it out.

**Comment by bsekachev** [comment]
@kirill-sizov Can you look to full-check run? Looks like it does not work for 3rd-party PRs.

**Comment by kirill-sizov** [comment]
@Marishka17 Have you already reviewed this PR? Or it requires additional review?

**Comment by Marishka17** [comment]
Unfortunately, I haven't, but that was planned. Need to review and test it with a container on Azure.

**Comment by nmanovic** [comment]
@Marishka17 , let's try to find sometime next week to review the PR.

**Comment by Marishka17** [comment]
@nmanovic, Ok, but in this case, I need an account on Azure for testing. I forgot to ask this again in our last meeting about this question.

File: cvat-ui/src/components/create-cloud-storage-page/cloud-storage-form.tsx [hunk 1]
```diff
@@ -414,6 +415,25 @@ export default function CreateCloudStorageForm(props: Props): JSX.Element {
...
+            return (
+                <>
+                    <Form.Item
+                        label='Connection string'
+                        name='connection_string'
+                        rules={[{ required: true, message: 'Please, specify your connection string' }]}
+                        {...internalCommonProps}
+                    >
+                        <Input.Password
+                            maxLength={437}
```
Comment (Marishka17) [inline cvat-ui/src/components/create-cloud-storage-page/cloud-storage-form.tsx:428]: Why is max length 437 here but in server serializer 440? What is the real maximum length for connection string?

File: cvat/apps/engine/cloud_provider.py [hunk 2]
```diff
@@ -382,10 +383,12 @@ class AzureBlobContainer(_CloudStorage):
     class Effect:
         pass
 
-    def __init__(self, container, account_name, sas_token=None):
+    def __init__(self, container, account_name, sas_token=None, connection_string=None):
```
Comment (Marishka17) [inline cvat/apps/engine/cloud_provider.py:386]: If I understand correctly, specifying `connection_string` is enough to connect to the original container, in this case, account_name should be an optional argument.

File: cvat/apps/engine/models.py [hunk 3]
```diff
@@ -733,6 +733,7 @@ class CredentialsTypeChoice(str, Enum):
     ACCOUNT_NAME_TOKEN_PAIR = 'ACCOUNT_NAME_TOKEN_PAIR' # nosec
     KEY_FILE_PATH = 'KEY_FILE_PATH'
     ANONYMOUS_ACCESS = 'ANONYMOUS_ACCESS'
+    CONNECTION_STRING = 'CONNECTION_STRING'
```
Comment (Marishka17) [inline cvat/apps/engine/models.py:736]: Could you please add migration? `python manage.py makemigrations`

File: cvat/apps/engine/serializers.py [hunk 4]
```diff
@@ -1210,8 +1211,8 @@ def validate_specific_attributes(self, value):
     def validate(self, attrs):
         provider_type = attrs.get('provider_type')
         if provider_type == models.CloudProviderChoice.AZURE_CONTAINER:
-            if not attrs.get('account_name', ''):
-                raise serializers.ValidationError('Account name for Azure container was not specified')
+            if not attrs.get('account_name', '') and not attrs.get('connection_string', ''):
+                raise serializers.ValidationError('Account name or Connection string for Azure container was not specified')
```
Comment (Marishka17) [inline cvat/apps/engine/serializers.py:1215]:
```suggestion
                raise serializers.ValidationError('Account name or сonnection string for Azure container was not specified')
```

File: site/content/en/docs/manual/basics/attach-cloud-storage.md [hunk 5]
```diff
@@ -277,6 +277,9 @@ and fill out the following form:
           in this case, you only need the storage account name to gain anonymous access.
           - `Account name` - storage account name.
 
+        - [`Connection string`](https://docs.microsoft.com/en-us/azure/storage/common/storage-configure-connection-string) -
+          - `Connection String` - connection string.
```
Comment (Marishka17) [inline site/content/en/docs/manual/basics/attach-cloud-storage.md:281]:
```suggestion
          - `Connection String` - A connection string which defines the resource being accessed and its authentication information.
```

**Review by Marishka17 — DISMISSED** [review]
@suzusuzu, Thanks for the contribution!
Could you please fix several minor comments and also update [`cvat-core/src/cloud-storage.ts`](https://github.com/opencv/cvat/blob/develop/cvat-core/src/cloud-storage.ts)  implementation by adding a connection string like other attributes implementation? (otherwise, the connection string will not be transferred from the client to the server)

**Comment by nmanovic** [comment]
@SpecLad , your check is working :). Should @Marishka17 just update it in the case?

File: cvat/schema.yml [hunk 6]
```diff
@@ -1,7 +1,7 @@
 openapi: 3.0.3
 info:
   title: CVAT REST API
-  version: 2.5.dev20230318195403
+  version: '2.5'
```
Comment (zhiltsov-max) [inline cvat/schema.yml]: Probably, this schema is incorrect. I suppose it was generated from the local environment, could you please try to generate it from the server image?

Comment (Marishka17) [reply cvat/schema.yml]: I've already re-generated it with server image

**Pull request merged**